File: app/src/main/java/de/dreier/mytargets/utils/DistanceInputDialog.java
/*
 * Copyright (C) 2017 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.utils;

import android.app.AlertDialog;
import android.content.Context;
import android.databinding.DataBindingUtil;
import android.text.InputType;
import android.view.LayoutInflater;
import android.widget.EditText;

import de.dreier.mytargets.R;
import de.dreier.mytargets.databinding.DialogCommentBinding;

public class DistanceInputDialog {

    public interface OnClickListener {
        void onOkClickListener(String input);
    }

    public static class Builder {
        private final Context mContext;
        private OnClickListener mClickListener;
        private String mUnit;

        public Builder(Context context) {
            mContext = context;
        }

        public void show() {
            LayoutInflater inflater = LayoutInflater.from(mContext);
            DialogCommentBinding binding = DataBindingUtil.inflate(inflater, R.layout.dialog_comment, null, false);
            binding.shotComment.setInputType(InputType.TYPE_CLASS_NUMBER);
            binding.unit.setText(mUnit);
            final EditText shotComment = binding.shotComment;

            new AlertDialog.Builder(mContext)
                    .setTitle(R.string.distance)
                    .setView(binding.getRoot())
                    .setPositiveButton(android.R.string.ok, (dialog, which) -> {
                        String s = shotComment.getText().toString();
                        mClickListener.onOkClickListener(s);
                        dialog.dismiss();
                    })
                    .setNegativeButton(android.R.string.cancel, (dialog, which) -> dialog.dismiss()).show();
        }

        public Builder setOnClickListener(OnClickListener listener) {
            mClickListener = listener;
            return this;
        }

        public Builder setUnit(String unit) {
            mUnit = unit;
            return this;
        }
    }
}

File: app/src/main/java/de/dreier/mytargets/features/distance/DistanceInputDialog.kt
/*
 * Copyright (C) 2018 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */

package de.dreier.mytargets.features.distance

import android.app.AlertDialog
import android.content.Context
import android.databinding.DataBindingUtil
import android.text.InputType
import android.view.LayoutInflater
import de.dreier.mytargets.R
import de.dreier.mytargets.databinding.DialogCommentBinding

class DistanceInputDialog {

    interface OnClickListener {
        fun onOkClickListener(input: String)
    }

    class Builder(private val context: Context) {
        private var clickListener: OnClickListener? = null
        private var unit: String? = null

        fun show() {
            val inflater = LayoutInflater.from(context)
            val binding = DataBindingUtil
                    .inflate<DialogCommentBinding>(inflater, R.layout.dialog_comment, null, false)
            binding.shotComment.inputType = InputType.TYPE_CLASS_NUMBER
            binding.unit.text = unit
            val shotComment = binding.shotComment

            AlertDialog.Builder(context)
                    .setTitle(R.string.distance)
                    .setView(binding.root)
                    .setPositiveButton(android.R.string.ok) { dialog, _ ->
                        val s = shotComment.text.toString()
                        clickListener?.onOkClickListener(s)
                        dialog.dismiss()
                    }
                    .setNegativeButton(android.R.string.cancel) { dialog, _ -> dialog.dismiss() }
                    .show()
        }

        fun setOnClickListener(listener: OnClickListener): Builder {
            clickListener = listener
            return this
        }

        fun setUnit(unit: String): Builder {
            this.unit = unit
            return this
        }
    }
}
